src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -628,13 +628,16 @@ public async Task HandleDidChangeTextDocumentNotification(ScriptFile[] changedFi
                 {
                     await Task.Run(() =>
                     {
+                        // Get the current ScriptInfo if one exists so we can lock it while we're rebuilding the cache
                         ScriptParseInfo scriptInfo = GetScriptParseInfo(connInfo.OwnerUri, createIfNotExists: false);
-                        if (scriptInfo != null && scriptInfo.IsConnected && 
+                        if (scriptInfo != null && scriptInfo.IsConnected &&
                             Monitor.TryEnter(scriptInfo.BuildingMetadataLock, LanguageService.OnConnectionWaitTimeout))
                         {
                             try
                             {
                                 this.BindingQueue.AddConnectionContext(connInfo, featureName: "LanguageService", overwrite: true);
+                                RemoveScriptParseInfo(rebuildParams.OwnerUri);
+                                UpdateLanguageServiceOnConnection(connInfo).Wait();
```

Comment:
Here's the only actual changes. I call UpdateLanguageServiceOnConnection to immediately rebuild the cache (it would do it the next time a request came in but this frontloads it so we can start that process early for slow connections)